features/steps/steps.rb

```diff
@@ -0,0 +1,69 @@
+Then("the Content-Type header is valid multipart form-data") do
+  expected = /^multipart\/form-data; boundary=--------------------------\d+$/
+  actual = Server.current_request[:request]["content-type"]
+
+  assert_match(expected, actual)
+end
+
+def read_expected_file(fixture, file_name)
+  path = "#{__dir__}/../expected/#{fixture}/#{file_name}"
+
+  assert_true(File.exists?(path), "The file '#{file_name}' does not exist at '#{path}'")
+
+  File.read(path)
+end
+
+def get_form_data_as_string(field)
+  form_data = read_key_path(Server.current_request[:body], field)
+
+  assert_instance_of(
+    WEBrick::HTTPUtils::FormData,
+    form_data,
+    "Expected the payload value for '#{field}' to be an instance of 'WEBrick::HTTPUtils::FormData' but it was a '#{form_data.class}'"
+  )
+
+  form_data.to_s
+end
+
+def assert_payload_fields_match(field, expected, actual)
+  result = value_compare(expected, actual)
```

Comment:
I realise this is extracted from the below functions but it feels slightly redundant, and I think I'd prefer assertions like this to be in the maze-runner code, or specific steps rather than floating functions.

I think it would make sense to add an standard assertion to the `CompareResult` class that can be called to make the below assertion in a standardised way, which would also be useful elsewhere.

Reply:
Maybe, but I'd prefer to pull it out separately to this PR if it's useful; otherwise we need to wait for a MR PR to be merged & released before this PR could be

Reply:
I agree on all of these points.  I'll create a separate PR to refactor when required.